offchain-modules/packages/scripts/src/utils/cardano_batch_test.ts

```diff
@@ -0,0 +1,125 @@
+import * as CardanoWasm from '@emurgo/cardano-serialization-lib-nodejs';
+import { asyncSleep } from '@force-bridge/x/dist/utils';
+import { logger } from '@force-bridge/x/dist/utils/logger';
+import CKB from '@nervosnetwork/ckb-sdk-core';
+import { WalletServer, Seed, AddressWallet, ShelleyWallet } from 'cardano-wallet-js';
+import { JSONRPCClient } from 'json-rpc-2.0';
+import fetch from 'node-fetch/index';
+import { prepareCkbPrivateKeys, prepareCkbAddresses, burn } from './eth_batch_test';
+
+export async function cardanoBatchTest(
+  ckbPrivateKey: string,
+  WALLET_SERVER_URL: string,
+  ckbNodeUrl: string,
+  ckbIndexerUrl: string,
+  forceBridgeUrl: string,
+  adaForceBridgeAddr: string,
+  adaWalletMnemonic: string,
+  batchNum = 100,
+  lockAmount = 2000000,
+  burnAmount = '1990000',
+): Promise<void> {
+  logger.info('adaBatchTest start!');
+  const ckb = new CKB(ckbNodeUrl);
+
+  const client = new JSONRPCClient((jsonRPCRequest) =>
+    fetch(forceBridgeUrl, {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+      },
+      body: JSON.stringify(jsonRPCRequest),
+      // id: 1,
+    }).then((response) => {
+      if (response.status === 200) {
+        // Use client.receive when you received a JSON-RPC response.
+        return response.json().then((jsonRPCResponse) => client.receive(jsonRPCResponse));
+      } else if (jsonRPCRequest.id !== undefined) {
+        return Promise.reject(new Error(response.statusText));
+      }
+    }),
+  );
+
+  logger.info('adaBatchTest adaWallet created');
+
+  const ckbPrivs = await prepareCkbPrivateKeys(batchNum);
+  const ckbAddresses = await prepareCkbAddresses(ckb, ckbPrivs, ckbPrivateKey, ckbNodeUrl, ckbIndexerUrl);
+
+  const passphrase = 'user_wallet_passphrase';
+  const adaWallet = await getUserWallet(WALLET_SERVER_URL, adaWalletMnemonic, passphrase);
+
+  const lockTxs = await lock(client, adaWallet, passphrase, ckbAddresses, adaForceBridgeAddr, lockAmount, 10 * 1000);
+  // await check(client, lockTxs, ckbAddresses, batchNum, adaTokenAddress);
```

Comment:
The CI failure reason is `generateBridgeOutNervosTransaction error Error: sudt amount is not enough!`

It means you don't have enough sudt ( which is Ada mirror  token on Nervos here ) to transfer out from Nervos.

In the log I found the mint tx have already been successful. There might be some mismatch between the mint tx and burn tx, either the sudt type or the recipient address.

The `check` uses the `getBridgeTransactionSummaries` method in rpc server to check whether the lock transaction succeeds. You should implement the function in RPC  server to make the history part in UI work. And we can take advantage of this in this test to check whether our lock tx succeeds.